cupy/_manipulation/add_remove.py

```diff
@@ -181,6 +192,8 @@ def unique(ar, return_index=False, return_inverse=False,
     mask = cupy.empty(aux.shape, dtype=cupy.bool_)
     mask[:1] = True
     mask[1:] = aux[1:] != aux[:-1]
+    if equal_nan:
+        _unique_update_mask_equal_nan(mask[1:], aux[1:], aux[:-1])
```

Comment:
Since nans are always after non-nan values, we don't have to use `aux[1:]`

https://numpy.org/doc/stable/reference/generated/numpy.sort.html
> In numpy versions >= 1.4.0 nan values are sorted to the end. The extended sort order is:
> > Real: [R, nan]
> > Complex: [R + Rj, R + nanj, nan + Rj, nan + nanj]

Reply:
Genius!